File: Leetcode-Easy/242. Valid Anagram/Valid Anagram.py
# https://leetcode.com/problems/valid-anagram/
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return sorted(s) == sorted(t)
        if(len(s)!=len(t)):
            return False
        countS, countT = {}, {}
        for i in range(len(s)):
            countS[s[i]] = 1 + countS.get(s[i],0)
            countT[t[i]] = 1 + countT.get(t[i],0)
        for c in countS:
            if (countS[c]!= countT.get(c,0)):
                return False
        return True
    

s = Solution()
print(s.isAnagram("anagram","nagaram"))

File: Leetcode-Easy/242. Valid Anagram/valid_anagram.java
// https://leetcode.com/problems/valid-anagram/

class Solution {

    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        Map<Character, Integer> m = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            m.put(s.charAt(i), m.get(s.charAt(i)) == null ? 1 : m.get(s.charAt(i)) + 1);
        }
        for (int i = 0; i < t.length(); i++) {
            m.put(t.charAt(i), m.get(t.charAt(i)) == null ? 1 : m.get(t.charAt(i)) - 1);
        }
        for (int val : m.values()) {
            if (val != 0) {
                return false;
            }
        }
        return true;

    }
    
}